incubator/hnc/api/v1alpha1/hierarchy_types.go

```diff
@@ -116,17 +116,22 @@ type Condition struct {
 	//
 	// Currently, the supported values are:
 	//
-	// - "CRIT_PARENT_MISSING": the specified parent is missing
+	// - "CritParentMissing": the specified parent is missing
 	//
-	// - "CRIT_PARENT_INVALID": the specified parent is invalid (e.g., would cause a cycle)
+	// - "CritParentInvalid": the specified parent is invalid (e.g., would cause a cycle)
 	//
-	// - "CRIT_ANCESTOR": a critical error exists in an ancestor namespace, so this namespace is no
+	// - "CritAncestor": a critical error exists in an ancestor namespace, so this namespace is no
 	// longer being updated either.
 	//
-	// - "REQUIRED_CHILD_CONFLICT": this namespace has a required child, but a namespace of the same
-	// name already exists and is not a child of this namespace. Note that the condition is _not_
-	// annotated onto the other namespace; it is considered an error _only_ for the would-be parent
-	// namespace.
+	// - "HNSMissing": this namespace is an owned namespace (created by reconciling an hns instance),
+	// but the hns instance is missing in its owner (referenced in its owner annotation) or the owner
```

Comment:
Will there be more information about which case (i.e., hns instance missing or owner namespace missing) causes this problem when a namespace has this error condition?

Reply:
No, we only look for the hns instance. If it doesn't exist, this condition will be set, no matter the owner namespace exists or not. The two cases here are actually the two possible cases of a missing hns instance.